### PR TITLE
Issue #616: Add route demotion to Step 3a when pr→patch

### LINKED ISSUE #616: auto: Step 3a 再判定で実行中ルートを M/pr → patch に縮退
## 背景

`/auto` の Step 3a（spec 後 Size リフレッシュ）が、Size を M→XS に再判定しても**実行中のルートは変更しない**。設計上は「次のルーティング判定にのみ適用、実行中フェーズシーケンスには遡及しない」と文書化されているが、これにより実コストが発生:

2026-06-14 のバッチ実行レポート（`docs/ja/reports/auto-batch-list-mode-2026-06-14.md`）で観測:

- #581: M→XS に再判定（ドキュメント only な ADR）。しかし `pr` ルートで code+review+merge に **38 分**支払い。patch ルートなら推定 10 分以内
- #548: 同様に M→XS に再判定。code+review+merge に約 33 分支払い

合計 60 分以上の無駄な実行時間。1 セッションあたり数件発生する典型パターンであり、累積コストは無視できない。

## 目的

spec フェーズで Size が M→XS または M→S に再判定された場合、残るフェーズ（code, review, merge）を patch ルートに動的に切り替える。これにより XS/S 化された Issue は patch ルートの所要時間（~10〜25 分）で完走する。

## 提案

`skills/auto/SKILL.md` の Step 3a を拡張:

```
Step 3a: Post-Spec Size Refresh（拡張案）

1. spec 完了後、Size を再フェッチ
2. ROUTE と REVIEW_DEPTH を更新（現状）
3. **【追加】** ROUTE が変更された場合（pr → patch）:
   a. 既存の Step 4 phase シーケンスを破棄
   b. patch ルートのシーケンスに切り替え: code(--patch) → verify
   c. ログ出力: "Post-spec route demotion: pr → patch, remaining phases re-planned"
4. 既存の Step 4 へ
```

トレードオフ:
- 複雑性増加: 実行中のフェーズシーケンス書き換え（既にコード生成済みの場合 worktree 状態の調整が必要かは要検討）
- 安全性: spec フェーズで Size 縮退判定が誤りだった場合、patch ルートが対応できないコード変更を起こす可能性

スコープ: Spec フェーズで明示的に Size を再判定したケースのみ対象（spec の出力が「documentation only」「single file edit」等のシグナルを示す場合）。

### Pre-merge

- [x]  Step 3a に route demotion 仕様が記述されている
- [x]  auto skill の bats テストが green（route demotion の新規ケース含む）

### Post-merge

- [x] 次回 `/auto` 実行で M→XS 再判定された Issue が patch ルート所要時間（< 25 分）で完走することを観察 

## 参照

- 2026-06-14 バッチレポート: `docs/ja/reports/auto-batch-list-mode-2026-06-14.md` の「再判定された Size は実行中のルートを変更しない」観察セクション
- 関連: #581, #548 が本問題に該当した実例

### PR DESCRIPTION
**Pull request by saito**
## Summary

- `skills/auto/SKILL.md` の Step 3a に route demotion 仕様を追加：pr → patch 縮退時に「Post-spec route demotion: pr → patch, remaining phases re-planned」をログ出力し、Step 4 では patch シーケンス（code --patch → verify）を使用するよう明示
- `tests/auto.bats` を新規作成：Step 3a セクションに "route demotion" および "Post-spec route demotion" が含まれることを検証する構造テスト（2件）
- `docs/structure.md`・`docs/ja/structure.md` のテストファイル数を 62 → 65 に更新

## Verification (pre-merge)

- `grep -A 20 "route demotion" "skills/auto/SKILL.md"` — route demotion 仕様が記述されていること
- `bats tests/auto.bats` — 新規 bats テスト 2件グリーン

## Verification (post-merge)

- 次回 `/auto` 実行で M→XS 再判定された Issue が patch ルート所要時間（< 25 分）で完走することを観察

closes #616

### CODE REVIEW

File: tests/auto.bats [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+SKILL_FILE="$(cd "$(dirname "$BATS_TEST_FILENAME")/.." && pwd)/skills/auto/SKILL.md"
+
+# Extract Step 3a section: from "### Step 3a:" to the next "### " heading
+step3a_section() {
+    awk '/^### Step 3a:/{found=1} found && /^### / && !/^### Step 3a:/{exit} found{print}' "$1"
+}
+
+@test "Step 3a section contains route demotion" {
+    run step3a_section "$SKILL_FILE"
```
Comment (saito) [inline tests/auto.bats:14]: **[Edge Case/Robustness] tests/auto.bats:14**

`run step3a_section "$SKILL_FILE"` の後に `[ "$status" -eq 0 ]` チェックがないため、SKILL_FILE が存在しない場合やawk失敗時にエラーが `$output` の文字列マッチ失敗として誤報告されます。

Severity: CONSIDER

Recommended fix:
```bash
@test "Step 3a section contains route demotion" {
    run step3a_section "$SKILL_FILE"
    [ "$status" -eq 0 ]
    [[ "$output" == *"route demotion"* ]]
}
```

**Review by saito — COMMENTED** [review]
## 受入条件検証結果

| 条件 | 結果 | 備考 |
|------|------|------|
| Step 3a に route demotion 仕様が記述されている | PASS | `grep -A 20 "route demotion" "skills/auto/SKILL.md"` — 仕様テキスト確認 |
| auto skill の bats テストが green | PASS | CI "Run bats tests" SUCCESS で代替検証 |
| M→XS 再判定 Issue が patch ルート所要時間で完走（観察） | POST-MERGE | observation event=auto-run — マージ後に観察 |

## CI ステータス

| ジョブ名 | ステータス | 備考 |
|---------|-----------|------|
| Run bats tests | SUCCESS | — |
| Validate skill syntax | SUCCESS | — |
| Forbidden Expressions check | SUCCESS | — |
| macOS shell compatibility | SUCCESS | — |

## コードレビュー

### 全体所見

Spec に定義された route demotion 仕様が `skills/auto/SKILL.md` Step 3a に正しく追加されており、`tests/auto.bats` による構造テスト2件も追加されています。CI はすべて SUCCESS。MUST 問題なし。

### General Comments（path/line 特定不可の指摘）

**[Spec Deviation] skills/auto/SKILL.md — Step 4 に ROUTE ディスパッチ宣言がなく、route demotion との接続が暗黙的**

- path: null
- Severity: SHOULD

Step 3a には「In Step 4, use the patch route sequence instead of the pr route sequence」と明示されており、Step 4 には "patch route XS/S (2 phases)" と "pr route (4 phases)" の各セクションが存在します。`ROUTE=patch` となれば patch セクションが適用されるため機能的には完全です。

ただし、Step 4 自体には「ROUTE 変数の値に基づいてどのセクションを選択するか」を明示するディスパッチ記述がなく、接続が暗黙的です。code フェーズのハンドオフでも「Step 4 の pr/patch ルート分岐コードは変更なし — 既存ロジックで対応済みか要確認」と deferred になっていた点です。

将来 Step 4 を編集した際に依存関係が断ち切れるリスクがあります。

Recommended fix:
Step 4 の冒頭（ルートセクション前）に次のような記述を追加する:
> "Select the route section below based on the current ROUTE value (set in Step 2 and potentially overridden by Step 3a's route demotion logic)."

---

### Issue Count Summary

- ラインコメントとして投稿: 1件 (MUST: 0, SHOULD: 0, CONSIDER: 1)
- General Comments（本文内）: 1件 (SHOULD: 1)

---
Next: SHOULD 対応後に `/merge 620` を実行。SHOULD を skip する場合はそのまま `/merge 620` へ。

**Comment by saito** [comment]
## Review Response Summary


### Claude Review Response

| Aspect | Filename | Line | Issue | Response | Fix/Skip reason |
|--------|---------|------|-------|----------|----------------|
| Spec Deviation | skills/auto/SKILL.md | — | Step 4 に ROUTE ディスパッチ宣言がなく route demotion との接続が暗黙的 | Resolved | Step 4 冒頭に「Select the route section below based on the current ROUTE value」を追加 |
| Edge Case/Robustness | tests/auto.bats | 14 | `run` 後に `[ "$status" -eq 0 ]` チェックなし | Skipped | CONSIDER レベルかつ実運用での障害可能性は低い（SKILL_FILE は固定パス）。別 Issue で改善を検討 |

### Lightweight Re-check Results

- validate-skill-syntax.py: PASS (0 errors, 0 warnings)
- Fix diff check: PASS
- New issues: none
